MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -807,11 +807,42 @@ private bool RequiresPostFiltering(InternalItemsQuery query)
             return false;
         }
 
+        private IReadOnlyList<BaseItem> SortItemsByRequest(InternalItemsQuery query, IReadOnlyList<BaseItem> items)
+        {
+            var ids = query.ItemIds;
+            int size = items.Count;
+
+            // ids can potentially contain non-unique guids, but query result cannot,
+            // so we include only first occurrence of each guid
+            var positions = new Dictionary<Guid, int>(size);
+            int index = 0;
+            for (int i = 0; i < ids.Length; i++)
+            {
+                if (positions.TryAdd(ids[i], index))
+                {
+                    index++;
+                }
+            }
+
+            var newItems = new BaseItem[size];
+            foreach(var item in items)
```

Comment:
```suggestion
            foreach (var item in items)
```
If you want ever better perf, change this into a `for` loop

Reply:
I've replaced `foreach` with `for`